lib/location/location_core.c

```diff
@@ -280,22 +288,30 @@ void location_core_config_log(const struct location_config *config)
 
 		if (type == LOCATION_METHOD_GNSS) {
 			LOG_DBG("      Timeout: %dms", config->methods[i].gnss.timeout);
+#if defined(CONFIG_LOG)
```

Comment:
I guess we should remove these compilation conditions that are added to get real coverage metrics from unit tests.
@tokangas, what do you think?

Reply:
I don't think those are very nice. Is the purpose of these to be able to flag functions `location_core_gnss_accuracy_str()` and `location_core_service_str()`? One possibility would be to test those functions directly from UT, I guess.

BTW, I think `log_strdup()` is not needed for strings returned by `location_core_gnss_accuracy_str()` and `location_core_service_str()` because the strings are constant.

Reply:
Is the purpose of these to be able to flag functions `location_core_gnss_accuracy_str()` and `location_core_service_str()`? One possibility would be to test those functions directly from UT, I guess.

Yes, enables flagging those functions out. Testing those functions requires us to make them non-static which is not perfect but better than this flagging. I don't know if I'll do these tests now but I'll certainly remove this flagging from the code.

> BTW, I think `log_strdup()` is not needed for strings returned by `location_core_gnss_accuracy_str()` and `location_core_service_str()` because the strings are constant.

True. Removing.